lib/account.rb

```diff
@@ -0,0 +1,66 @@
+require 'csv'
+
+module Bank
+  class Account
+    attr_reader :id, :balance, :open_date
+
+    def initialize(id, balance, open_date = nil)
+
+      raise ArgumentError.new("balance must be >= 0") if balance < 0
+      @id = id
+      @balance = balance
+      @open_date = open_date
+    end
+
+    def self.all #better name! .create_all_accounts_from_csv
```

Comment:
It depends on your application. We'll see this pattern a lot in Rails once we start using ActiveRecord and databases.